Mathlib/GroupTheory/GroupAction/FixedPoints.lean

```diff
@@ -0,0 +1,292 @@
+  ext x
+  rw [mem_fixedBy, mem_fixedBy, inv_smul_eq_iff, eq_comm]
+
+@[to_additive]
+theorem smul_mem_fixedBy_iff_mem_fixedBy {a : α} {g : G} :
+    g • a ∈ fixedBy α g ↔ a ∈ fixedBy α g := by
+  rw [mem_fixedBy, smul_left_cancel_iff]
+  rfl
+
+@[to_additive]
+theorem smul_inv_mem_fixedBy_iff_mem_fixedBy {a : α} {g : G} :
+    g⁻¹ • a ∈ fixedBy α g ↔ a ∈ fixedBy α g := by
+  rw [← fixedBy_inv_eq_fixedBy, smul_mem_fixedBy_iff_mem_fixedBy, fixedBy_inv_eq_fixedBy]
+
+@[to_additive minimalPeriod_eq_one_of_fixedBy]
+theorem minimalPeriod_eq_one_of_fixedBy {a : α} {g : G} (a_in_fixedBy : a ∈ fixedBy α g) :
+    Function.minimalPeriod (fun x => g • x) a = 1 := by
+  rwa [Function.minimalPeriod_eq_one_iff_isFixedPt]
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_subset_fixedBy_zpow (g : G) (j : ℤ) :
+    fixedBy α g ⊆ fixedBy α (g ^ j) := by
+  intro a a_in_fixedBy
+  rw [mem_fixedBy, zpow_smul_eq_iff_minimalPeriod_dvd,
+    minimalPeriod_eq_one_of_fixedBy a_in_fixedBy, Nat.cast_one]
+  exact one_dvd j
+
+variable (M α) in
+@[to_additive (attr := simp)]
+theorem fixedBy_one_eq_univ : fixedBy α (1 : M) = Set.univ :=
+  Set.eq_univ_iff_forall.mpr (fun a => one_smul M a)
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_mul (m₁ m₂ : M) : fixedBy α m₁ ∩ fixedBy α m₂ ⊆ fixedBy α (m₁ * m₂) := by
+  intro a ⟨h₁, h₂⟩
+  rw [mem_fixedBy, mul_smul, h₂, h₁]
+
+variable (α) in
+@[to_additive]
+theorem fixedBy_conj (g h : G) :
+    fixedBy α (h * g * h⁻¹) = (fun a => h⁻¹ • a) ⁻¹' fixedBy α g := by
+  ext a
+  rw [Set.mem_preimage, mem_fixedBy, mem_fixedBy]
+  repeat rw [mul_smul]
+  exact smul_eq_iff_eq_inv_smul h
+
+variable (α) in
+@[to_additive]
+theorem smul_fixedBy (g h: G) :
+    h • fixedBy α g = fixedBy α (h * g * h⁻¹) := by
+  rw [fixedBy_conj, Set.preimage_smul, inv_inv]
+
+end FixedPoints
+
+section Pointwise
+
+/-!
+### `fixedBy` sets of the pointwise group action
+
+The theorems below need the `Pointwise` scoped to be opened (using `open Pointwise`)
+to be used effectively.
+-/
+
+/--
+If a set `s : Set α` is in `fixedBy (Set α) g`, then all points of `s` will stay in `s` after being
+moved by `g`.
+-/
+@[to_additive "If a set `s : Set α` is in `fixedBy (Set α) g`, then all points of `s` will stay in
+`s` after being moved by `g`."]
+theorem set_mem_fixedBy_iff (s : Set α) (g : G) :
+    s ∈ fixedBy (Set α) g ↔ ∀ x, x ∈ s ↔ g • x ∈ s := by
+  constructor
+  · intro fixed x
+    refine ⟨
+      fun x_in_s => fixed ▸ Set.smul_mem_smul_set x_in_s,
+      fun gx_in_s => ?inv_in_set⟩
+    rwa [← fixed, Set.smul_mem_smul_set_iff] at gx_in_s
+  · intro closed
+    rw [← fixedBy_inv_eq_fixedBy]
+    ext x
+    rw [Set.mem_inv_smul_set_iff]
+    exact ⟨fun gxs => (closed x).mpr gxs, fun xs => (closed x).mp xs⟩
+
+theorem smul_mem_of_set_mem_fixedBy {s : Set α} {g : G} (s_in_fixedBy : s ∈ fixedBy (Set α) g)
+    {x : α} : x ∈ s ↔ g • x ∈ s := (set_mem_fixedBy_iff s g).mp s_in_fixedBy x
```

Comment:
Usually we would put the more complicated expression on the left side of the `iff`, so that it `rw`'s naturally.